### PR TITLE
update connect/ca's vault AuthMethod conf section

### PR DESCRIPTION
**Pull request by eikenb**
Added list of supported types to Type field.

Updated Params field to re-frame as supporting arguments specific to the supported vault-agent auth-auth methods with links to each methods "#configuration" section.

Included a Note box to call out limits on parameters supported.

--

Note that this leaves out anything about how the old Params worked. It was a hack that let practitioners hardcode auth credentials (eg. the JWT token string). This document update is about the new functionality where the configuration options for Vault Agent's auto-auth methods are directly supported and it works in a proper manner (ie. no more hardcoded tokens in the config).

### CODE REVIEW

File: website/content/docs/connect/ca/vault.mdx [hunk 1]
```diff
@@ -94,17 +94,14 @@ The key after the slash refers to the corresponding option name in the agent con
...
-    [Vault Auth Methods](/vault/docs/auth) for information on how to configure the
-    auth method you wish to use. If using the Kubernetes auth method,
-    Consul will read the service account token from the
-    default mount path `/var/run/secrets/kubernetes.io/serviceaccount/token` if the `jwt` parameter
-    is not provided.
+   - `Params`/`params` (`map: nil`) - The parameters to configure the auth method. The configuration parameters needed will depend on which auth type you are using. Please see the Vault Agent, auto-auth method documentation for details on their configuration options: [AppRole](https://developer.hashicorp.com/vault/docs/agent/autoauth/methods/approle#configuration), [AWS](https://developer.hashicorp.com/vault/docs/agent/autoauth/methods/aws#configuration), [Azure](https://developer.hashicorp.com/vault/docs/agent/autoauth/methods/azure#configuration), [GCP](https://developer.hashicorp.com/vault/docs/agent/autoauth/methods/gcp#configuration), [JWT](https://developer.hashicorp.com/vault/docs/agent/autoauth/methods/jwt#configuration), [Kubernetes](https://developer.hashicorp.com/vault/docs/agent/autoauth/methods/kubernetes#configuration).
```
Comment (im2nguyen) [inline website/content/docs/connect/ca/vault.mdx:102]:
```suggestion
   - `Params`/`params` (`map: nil`) - The parameters to configure the auth method. The configuration parameters needed will depend on which auth type you are using. Please refer to the Vault Agent auto-auth method documentation for details on their configuration options: [AppRole](/vault/docs/agent/autoauth/methods/approle#configuration), [AWS](/vault/docs/agent/autoauth/methods/aws#configuration), [Azure](/vault/docs/agent/autoauth/methods/azure#configuration), [GCP](/vault/docs/agent/autoauth/methods/gcp#configuration), [JWT](/vault/docs/agent/autoauth/methods/jwt#configuration), [Kubernetes](/vault/docs/agent/autoauth/methods/kubernetes#configuration).
```

File: website/content/docs/connect/ca/vault.mdx [hunk 2]
```diff
@@ -94,17 +94,14 @@ The key after the slash refers to the corresponding option name in the agent con
...
+
+   ~> **Note:** Only the authentication related fields are supported, not the optional management fields. E.g. JWT's `path` and `role` are supported but not its `remove_jwt_after_reading` field.
```
Comment (im2nguyen) [inline website/content/docs/connect/ca/vault.mdx:104]:
```suggestion
   Only the authentication related fields (for example, JWT's `path` and `role`) are supported. The optional management fields (for example: `remove_jwt_after_reading`) are not supported.
```

**Review by im2nguyen — APPROVED** [review]
this looks good!

**Comment by eikenb** [comment]
Thanks @im2nguyen!

**Pull request merged**